app/controllers/works_controller.rb

```diff
@@ -0,0 +1,101 @@
+class WorksController < ApplicationController
+  def index
+    @works = Work.all
+  end
+
+  def show
+    work_id = params[:id]
+    @work = Work.find_by(id: work_id)
+    if @work.nil?
+      flash[:warning] = "This particular work does not exist"
+      redirect_to works_path
+      return
+    end
+    render :show
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(work_params)
+    if @work.save # save returns true if the database insert succeeds
+      flash[:success] = "#{@work.category.capitalize} has been added successfully"
+      redirect_to work_path(@work.id) # go to the index so we can see the work in the list
+    else # save failed :(
+      flash.now[:error] = "Oops! Something went wrong. #{@work.category.capitalize} was not added"
+      render :new # show the new work form view again
+    end
+  end
+
+  def edit
+    work_id = params[:id]
+    @work = Work.find_by(id: work_id)
+    if @work.nil?
+      flash[:warning] = "This particular work does not exist"
+      redirect_to root_path
+      return
+    end
+  end
+
+  def update
+    if @work.nil?
```

Comment:
@work will always be nil here because it's scope doesn't carry over from the other methods from previous requests.
The same two lines you have at the top of the show action belong at the top of this action and the destroy action.
```suggestion
    work_id = params[:id]
    @work = Work.find_by(id: work_id)
    if @work.nil?
```